release-notes/opensearch-release-notes-2.7.0.md

```diff
@@ -0,0 +1,642 @@
+# OpenSearch and OpenSearch Dashboards 2.7.0 Release Notes
+
+## Release Highlights
+OpenSearch 2.7.0 includes a number of new and enhanced features for your search, observability, and analytics workloads. This release delivers production-ready functionality for segment replication, searchable snapshots, and the ability to add multiple data sources to a single dashboard—features that were previously released as experimental. Also included are new flat object fields for more efficient indexing, integrated observability features within OpenSearch Dashboards, new geospatial functionality, and updates designed to simplify administration of indexes and clusters.
+
```

Comment:
@zelinh could we add highlight here?

OpenSearch Dashboards version 2.7 has introduced an improvement where administrators can now override the NodeJs version used to run the software. This modification has the potential to affect the launch logic and may require changes to existing infrastructure in order to properly launch OpenSearch Dashboards. Further information about this change can be found at the following link: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3508

Reply:
@seraphjiang Just to clarify, by default there is no impact to normal OpenSearch users but adds additional capability for administrators to override NodeJs version used to run the software (when needed?). I see that there is a caveat stating that it has potential to affect the launch logic and may require changes to existing infrastructure in order to properly launch OpenSearch Dashboards when overriding it. Can we elaborate more on that rather than pointing to a PR?

Reply:
Basically, point to a documentation / readme file with the information needed for launch logic when overriding the NodeJs version used to run the software.

Reply:
The change is to allow user to choose NodeJS, the implementing has issue which is addressed in https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3955

For impacted customer, they has two options, modify their infrastructure code to adapt to the change, or apply above fix as workaround before we bring to release offically.

Reply:
@bbarani if we could not put in highlight, i would suggest to add 3508 to break changes. 

This break change is not breaking functionality or api interface, but impact the launch behavior, although it was for another enhancement purpose. 

I'll leave this to team to make decision is better for the best of interest for community. We found this issue in early adoption and test of 2.7.

Reply:
@jhmcintyre @seraphjiang Does it make sense to add a new "Known issues/bugs" section to release notes and list this there with the workaround?

Reply:
@bbarani @seraphjiang I would think adding a section called "Known issues" below the release highlights section would make more sense than categorizing this under "release highlights". (I don't know if I'd include "bugs" in that section title b/c this doesn't seem like a bug but open to feedback.

Reply:
Thanks @bbarani  and @jhmcintyre the known issues works. 
This could help other customer to find workaround if they are facing same issue.

Reply:
@seraphjiang can you point me to where the workaround is documented and how the admin would apply it? I'm looking at 3508 and it's not entirely clear (at least to me).

Reply:
The workaround is at https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3955/ not https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3508/

The instruction is to apply the same fix in 3955, and rebuild the OpenSearch Dashboards for the artifacts. For how to build , refer to https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#building-artifacts

Reply:
@seraphjiang @bbarani  taking a stab at writing this up. Is this accurate, and sufficiently comprehensive? 

Known Issues
 
Version 2.7.0 of OpenSearch Dashboards offers administrators the ability to manually override the version of Node.js used to run the software. If an administrator chooses this option, the change has the potential to affect the launch logic for OpenSearch Dashboards, in which case the administrator may need to modify their infrastructure code or apply a workaround to enable Dashboards to launch properly. The workaround involves making a change to the CLI shell scripts as described in this issue https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3955 and rebuilding the artifacts for Dashboards as described here https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#building-artifacts. If the administrator does not choose to override the version of Node.js used to run the software, this will not be an issue.

Reply:
Overall LGTM, except `If the administrator does not choose to override the version of Node.js used to run the software, this will not be an issue.` we might remove this as it will also impact cx not override the nodejs.

  thanks @jhmcintyre

Reply:
Thanks @seraphjiang and I'm fine removing that last line. @bbarani, does this work for you?